Wrapping/Generators/CastXML/CMakeLists.txt

```diff
@@ -60,6 +60,9 @@ else()
     if(ITK_BINARY_DIR)
       itk_download_attempt_check(CastXML)
     endif()
+    if(${CMAKE_VERSION} VERSION_LESS 3.24)
+      cmake_policy(SET CMP0135 NEW)
```

Comment:
I saw this change after my comment:
https://github.com/InsightSoftwareConsortium/ITK/blob/master/CMakeLists.txt#L21

Reply:
Yep, https://github.com/InsightSoftwareConsortium/ITK/pull/3709#issuecomment-1295966013 The question there is how to inherit those values if we think they should apply to the wrapping projects.

Reply:
This makes no sense. CMP0135 didn't exist until 3.24. I think you want `if (POLICY CMP0135)` to detect whether the policy exists or not.

You probably want a `cmake_policy(PUSH)` scope (with corresponding `POP` later).

Reply:
> This makes no sense. CMP0135 didn't exist until 3.24.

Yes, Ben, you're right, I realized yesterday when I still saw the CI warning.

> I think you want if (POLICY CMP0135) to detect whether the policy exists or not.
> You probably want a cmake_policy(PUSH) scope (with corresponding POP later).

Can you elaborate on this please? As you see in the comment above, the policy is set for the root `CMakeLists.txt` so I am wondering whether your solution would allow us to inherit or apply existing policies to other subprojects such as the one here. According to the [documentation](https://cmake.org/cmake/help/latest/command/cmake_policy.html#cmake-policy-stack) we may be willing to use `block()/endblock()`?

Reply:
Policies have their own scoping mechanism, so you'll want to push and pop any policies you set in something like this.

Reply:
Sorry, Ben, maybe I was not clear in my question. ITK sets the following policies:
https://github.com/InsightSoftwareConsortium/ITK/blob/master/CMakeLists.txt#L21

So if I understand well, according to the CMake [documentation](https://cmake.org/cmake/help/latest/command/cmake_policy.html#cmake-policy-stack), they are put on a stack. So the question is if doing the below
```
cmake_policy(PUSH)
ExternalProject_Add(castxml
  URL ${_castxml_url}
  URL_HASH SHA512=${_castxml_hash}
  CONFIGURE_COMMAND ""
  BUILD_COMMAND ""
  INSTALL_COMMAND "${CMAKE_COMMAND}" -E copy_directory "${CMAKE_CURRENT_BINARY_DIR}/castxml-prefix/src/castxml" "${CMAKE_CURRENT_BINARY_DIR}/castxml"
  )
cmake_policy(POP)
```

would apply the values of ALL policies in the stack (so `CMP0135` to `NEW` among them) to the project we are configuring within the `cmake_policy(PUSH)/cmake_policy(POP)` scope.

Reply:
The policy would be set for the `ExternalProject_add` call which basically does some `add_custom_command` calls at heart. By the time `castxml` is running its configure step, the policy scope mentioned here is long gone and doesn't do anything.

Reply:
OK, that confuses me further then, as you suggested using `cmake_policy(PUSH)/cmake_policy(POP)` in https://github.com/InsightSoftwareConsortium/ITK/pull/3709#discussion_r1027296475.

Should we just do ?
```
if(POLICY CMP0135)
      cmake_policy(SET CMP0135 NEW)
```

Reply:
Yes, if you're setting policies not at top-level, use a scope:

```cmake
cmake_policy(PUSH)
if (POLICY CMP0135)
  cmake_policy(SET CMP0135 NEW)
endif ()
# Code that needs CMP0135
cmake_policy(POP)
```

It's just that the policy does *not* affect the project ExternalProject is building, only the prep work that EP itself does.

Reply:
OK, thanks for the additional explanation Ben. Unless you tell the contrary, I think it is fine that the policy applies to the EP prep work, as the step that is triggering the warning is related to the moment the files are extracted.

I've amended the commit. Let's see if that makes the things work.

Reply:
Resolving as changes have been addressed. Unresolve in case you still find that the issues were not addressed appropriately.